src/mlpack/core/math/random.hpp

```diff
@@ -58,6 +58,27 @@ inline double Random()
 }
 
 /**
+ * Generate a random index within [0, N) under the given distribution,
+ * where N is equal to the size of the distribution.
+ *
+ * @param distribution The probability distribution of each index, user must
+ *   make sure the sum of its elements is 1.
+ * @return the random index.
+ */
+inline size_t RandomIndex(const arma::colvec& distribution)
+{
+  double oops = Random();
+  double density = distribution(0);
+  size_t index = 0;
+  while (oops >= density)
+  {
+    index++;
+    density += distribution(index);
+  }
+  return index;
+}
```

Comment:
It's amazing, I even don't know that class. And I find a better solution, we can use `std:: discrete_distribution`